net/portmapper/portmapper.go

```diff
@@ -401,18 +420,31 @@ func (c *Client) createOrGetMapping(ctx context.Context) (external netaddr.IPPor
 
 	pmpAddr := netaddr.IPPortFrom(gw, pmpPort)
 	pmpAddru := pmpAddr.UDPAddr()
+	pcpAddr := netaddr.IPPortFrom(gw, pcpPort)
+	pcpAddru := pcpAddr.UDPAddr()
 
-	// Ask for our external address if needed.
-	if m.external.IP().IsZero() {
-		if _, err := uc.WriteTo(pmpReqExternalAddrPacket, pmpAddru); err != nil {
+	preferPCP := DisablePMP || (!haveRecentPMP && haveRecentPCP)
```

Comment:
shouldn't need to check DisablePMP here? If we didn't send the PMP solicitation out before, haveRecentPMP will be false, right?

But if we do need it, why don't we also need to check DisablePCP?

Reply:
I'm not sure what you mean by PMP solicitation, other than the the request for the external IP, but I don't think that will update haveRecentPMP

I updated it to make sure that it won't try PCP if PCP is disabled